src/store/traction/pacbio/requests/actions.js

```diff
@@ -87,7 +76,7 @@ const sampleExtractionTubeJson = (tubes) => {
         library_type,
         estimate_of_gb_required,
         number_of_smrt_cells,
-        cost_code: processCostCode(t),
+        ...(t.cost_code === null ? null : { cost_code: t.cost_code }),
```

Comment:
I think all you need to do is make cost_code part of the mapping i.e. 

```
return tubes.map(
    ({
      library_type,
      estimate_of_gb_required,
      number_of_smrt_cells,
      barcode,
      study_uuid: external_study_id,
      sample_uuid: external_id,
      cost_code,
      fields: { sanger_sample_id: name, sample_common_name: species },
      ...t
    }) => ({
      sample: { name, species, external_id },
      request: {
        external_study_id,
        library_type,
        estimate_of_gb_required,
        number_of_smrt_cells,
        cost_code,
      },
      tube: { barcode },
    }),
  )
```

Reply:
The spread operator for assigning the cost code is needed because we dont want to include the cost code into the request if it is null /empty. This is because the backend will not default null values it only defaults if the value doesnt exist. So in this case it checks if the cost code is null and only includes it into the request if it is not null

Reply:
OK. But we can still simplify it a bit. Like so:

```
return tubes.map(
    ({
      library_type,
      estimate_of_gb_required,
      number_of_smrt_cells,
      barcode,
      study_uuid: external_study_id,
      sample_uuid: external_id,
      cost_code,
      fields: { sanger_sample_id: name, sample_common_name: species },
      ...t
    }) => ({
      sample: { name, species, external_id },
      request: {
        external_study_id,
        library_type,
        estimate_of_gb_required,
        number_of_smrt_cells,
        ...(cost_code ? {cost_code} : null)
      },
      tube: { barcode },
    }),
  )

```

Reply:
👍